include/parlay/utilities.h

```diff
@@ -38,11 +59,31 @@ template<typename T>
 #endif
 }
 
+// Given storage containing valid bits for object of type T, produce an object of type T
+// with the same object representation.
+//
+// i.e., basically std::bit_cast from C++20 but with a slightly different interface. The
+// goal is to type pun from a byte representation into a valid object with valid lifetime.
+//
+// Note the subtle difference between from_bytes (above) and bits_to_object. from_bytes(p)
+// assumes that an object of type T with valid lifetime *already exists* at the memory
+// location p, and you just want to get your hands on it, while bits_to_object assumes
+// that src contains a byte pattern that represent a T object, but no such T object
+// actually exists at that location. bits_to_object creates a copy of the bytes at src
+// and starts the lifetime of a valid object of type T (assuming T is trivially copyable).
+template<typename T>
+PARLAY_INLINE T bits_to_object(const char* src) {
+  if constexpr (!std::is_trivially_default_constructible_v<T>) {
+    uninitialized<T> holder;
+    std::memcpy(&holder.value, src, sizeof(T));
+    return holder.value;
```

Comment:
This seems like overengineering. IIUC the whole point of this `bits_to_object` is you're trying to avoid the "move and destroy" codepath in `relocate` below. But `return holder.value` won't copy-elide, so you'll get a move anyway. In fact you'll get a _copy_ anyway, won't you?!
So unless I'm mistaken, this is a pessimization and also must be lacking test coverage for `unique_ptr` or any other trivially-relocatable-but-non-copyable type.